packages/core/src/utils/useControlled.ts

```diff
@@ -2,21 +2,21 @@ import {
   Dispatch,
   isValidElement,
   SetStateAction,
+  useCallback,
+  useEffect,
   useRef,
   useState,
-  useEffect,
-  useCallback,
 } from "react";
 
-export interface UseControlledProps<T = unknown> {
+export interface UseControlledProps<T> {
   /**
    * Holds the component value when it's controlled.
    */
-  controlled: T | undefined;
+  controlled?: T;
   /**
    * The default value when uncontrolled.
    */
-  default: T | undefined;
+  default: T;
```

Comment:
is `default` mandatory now?

Reply:
Sort of, the value for default is mandatory, but it could still be undefined. Default should always exist for the uncontrolled scenario and passing in the controlled prop will always override it, so there's no behaviour change here.